src/sprites/entities/entity.py

```diff
@@ -0,0 +1,221 @@
+
+    def get_facing_direction(self):
+        self.facing_direction = get_entity_facing_direction(
+            self.direction, self.facing_direction
+        )
+        self._update_axe_hitbox()
+
+    def get_target_pos(self):
+        return screen_to_tile(self.hitbox_rect.center)
+
+    def focus(self):
+        self.focused = True
+        self.focused_indicator = Sprite(
+            (0, 0), indicators.ENTITY_FOCUSED, (self.groups()[0],),
+            Layer.EMOTES
+        )
+
+    def unfocus(self):
+        self.focused = False
+        if self.focused_indicator:
+            self.focused_indicator.kill()
+            self.focused_indicator = None
+
+    @abstractmethod
+    def move(self, dt: float):
+        pass
+
+    def check_collision(self):
+        """
+        :return: true: Entity collides with a sprite in self.collision_sprites,
+        otherwise false
+        """
+        colliding_rect = None
+
+        for sprite in self.collision_sprites:
+            if sprite is not self:
+
+                if sprite.hitbox_rect.colliderect(self.hitbox_rect):
+                    colliding_rect = sprite.hitbox_rect
+                    distances_rect = colliding_rect
+
+                    if isinstance(sprite, Entity):
+                        # When colliding with another entity, the hitbox to
+                        # compare to will also reflect its last-frame's state
+                        distances_rect = sprite.last_hitbox_rect
+
+                    # Compares each point of the last-frame's hitbox to the
+                    # hitbox the Entity collided with, to check at which
+                    # direction the collision happened first
+                    distances = (
+                        abs(self.last_hitbox_rect.right - distances_rect.left),
+                        abs(self.last_hitbox_rect.left - distances_rect.right),
+                        abs(self.last_hitbox_rect.bottom - distances_rect.top),
+                        abs(self.last_hitbox_rect.top - distances_rect.bottom)
+                    )
+
+                    shortest_distance = min(distances)
+                    if shortest_distance == distances[0]:
+                        self.hitbox_rect.right = colliding_rect.left
+                    elif shortest_distance == distances[1]:
+                        self.hitbox_rect.left = colliding_rect.right
+                    elif shortest_distance == distances[2]:
+                        self.hitbox_rect.bottom = colliding_rect.top
+                    elif shortest_distance == distances[3]:
+                        self.hitbox_rect.top = colliding_rect.bottom
+
+        self.is_colliding = bool(colliding_rect)
+
+    @abstractmethod
+    def animate(self, dt: float):
+        """
+        Animate the Entity. Child classes should implement method and
+        set current image based on self._current_ani_frame
+        """
+        self.frame_index += 4 * dt
+
+    def prepare_for_update(self):
```

Comment:
If this is not going to be used outside of `Entity`, it's probably better to make this method protected.

```suggestion
    def _prepare_for_update(self):
```